### PR TITLE
Add uncontroversial RBF base features

### PR DESCRIPTION
**Pull request by jonasschnelli**
- add option to set nSequence in createrawtransaction
- add nSequence option to bitcoin-tx
- Refactor `CreateTransaction` (remove sign-bool and make use of bit flags)

### CODE REVIEW

**Comment by jonasschnelli** [comment]
The uncontroversial commits of #7159.
Is unclear how we allow opt-in-RBF over RPC and in the GUI.
One proposal would be to add a command `alterrawtransaction` (that could support setting the opt-in nSequence for all inputs of the provided rawtx. The command could also take care of increasing the fees).

**Comment by dcousens** [comment]
Was https://github.com/bitcoin/bitcoin/pull/7159 controversial?

**Comment by jonasschnelli** [comment]
@dcousens Yes. #7159 contains booleans for `createrawtransaction` and `fundrawtransaction` to opt-int-RBF. Not sure if we want this or if we should go with a new call (something like `alterrawtransaction`). Also the process/workflow how to change the fee of a opt-in-RBF transaction is unclear (could also go over `alterrawtransaction`). And then, there is also the point where we once said we should move the utility-only functions to `bitcoin-tx`, etc.

This PR does only add a feature to set the nSequence number per input for `createrawtransaction` and `bitcoin-tx`. This is not directly related to RBF.

At least with this PR someone can create a RBF tx without fiddling with the hex-string (good for testing future changes as well).

**Comment by jonasschnelli** [comment]
Closing in favor of #7865.

File: src/rpc/rawtransaction.cpp [hunk 1]
```diff
@@ -334,6 +334,7 @@ UniValue createrawtransaction(const UniValue& params, bool fHelp)
             "       {\n"
             "         \"txid\":\"id\",    (string, required) The transaction id\n"
             "         \"vout\":n        (numeric, required) The output number\n"
+            "         \"sequence\":n    (numeric, optional) The sequence number (will overwrite RBF opt-in)\n"
```
Comment (gmaxwell) [inline src/rpc/rawtransaction.cpp]: Perhaps the will override comment is a bit confusing with the functionality missing?   maybe add a "in future"?

**Pull request closed without merging**